### DISCUSSION TITLE
How to filter the deleted entities?

### DISCUSSION THREAD

**Discussion by ladiesman1987** [mozman/ezdxf#439] [Q&A]
I use the purge() method to avoid getting deleted entities, like:
`msp.purge()`
but it shows the modelspace doesn't have this method.
so ,how can i filter the deleted entities?

**Comment by mozman** [comment]
When iterating over an layout, it does not yield deleted entities e.g. `for e in msp: ...`. If you have to purge the deleted entities manually use: `msp.entity_space.purge()`. The entities are removed automatically from the layout if you delete the entities by the safer option `msp.delete_entity(e)`.

**Reply by mozman** [reply]
added `BaseLayout.purge()` to avoid access to the layout internals, available in v0.16.3.

**Reply by ladiesman1987** [reply]
Thanks for your advice, I have solved this problem.
